XamarinApp/XamarinApp/ViewModels/LoginPageViewModel.cs

```diff
@@ -0,0 +1,135 @@
+using Prism.Commands;
+using Prism.Navigation;
+using Prism.Services;
+using System;
+using Xamarin.Forms;
+using XamarinApp.Models;
+using XamarinApp.PageName;
+using XamarinApp.Resx;
+using XamarinApp.Services.Interfaces;
+
+namespace XamarinApp.ViewModels
+{
+  public class LoginPageViewModel : ViewModelBase
+  {    
+    private readonly ILogManager _logManager;
+    private readonly ILogger _logger;
+    private readonly INavigationService _navigation;
+    private readonly IPageDialogService _pageDialogService;
+    private readonly IUserLoginService _userLoginService;
+
+    public DelegateCommand LoginCommand { get; set; }
+    public DelegateCommand SignUPCommand { get; set; }
+    private string _userName;
+    private string _password;
+
+    private string _name;
+
+    public string Name
+    {
+      get { return _name; }
+      set { SetProperty(ref _name, value); }
+    }
+
+    public string UserName
+    {
+      get
+      {
+        return _userName;
+      }
+      set
+      {
+        SetProperty(ref _userName, value);
+      }
+    }
+    public string PassWord
+    {
+      get
+      {
+        return _password;
+      }
+      set
+      {
+        SetProperty(ref _password, value);
+      }
+    }
+    public LoginPageViewModel(INavigationService navigationService, IPageDialogService pageDialogService,
+      IUserLoginService userLoginService, ILogger logger, ILogManager logManager)
+    {
+      _navigation = navigationService;
+      _pageDialogService = pageDialogService;
+      _userLoginService = userLoginService;
+      _logger = logManager.GetLog();
+      _logManager = logManager;
+      LoginCommand = new DelegateCommand(OnLoginClicked);
+      SignUPCommand = new DelegateCommand(OnSignUpClicked);
+    }
+
+    private async void OnSignUpClicked()
+    {
+      try
+      {
+        IsLoading = true;
+        IndicatorVisible = true;
+        var answer = await _pageDialogService.DisplayAlertAsync(AppResource.Request, AppResource.PageDialogRequest, "Yes", "No");
+        if (answer)
+        {
+          await _navigation.NavigateAsync(PageNames.SignUpPage);          
+        }
+      }
+
+      catch (Exception ex)
+      {
+        await _pageDialogService.DisplayAlertAsync(AppResource.Alert, ex.Message, "Ok");
+      }
+
+      finally
+      {
+        IsLoading = false;
+        IndicatorVisible = false;
+      }
+    }
+    public async void OnLoginClicked()
```

Comment:
Don't write async void use only async Task everywhere.

Read why async void is bad - https://stackoverflow.com/questions/45447955/why-exactly-is-void-async-bad